### DISCUSSION TITLE
Limited number of high-performance GitHub-hosted runners?

### DISCUSSION THREAD

**Discussion by selfagency** [community/community#37570] [Actions]
I'm in the high-performance GitHub-hosted runners public beta and...I can maybe run 5 simultaneous workflows using larger runners. During the business day, my team could have 30+ workflows going at once. This won't do at all if these kinds of waits are what I have to look forward to.

![CleanShot 2022-10-27 at 18 57 48](https://user-images.githubusercontent.com/2541728/198413278-5c1e44ec-2274-4a4d-a78a-2d0e2c2ffb37.png)
Your intro email says:

> With new, high-performance GitHub-hosted runners, teams can reduce bottlenecks

This is _already_ a bottleneck I'm going to have to revert.

**Comment by airtonix** [comment]
Experiencing something that I think is related to this.

I noticed a banner that said something along the lines of "big runners... now enabled for your org"... yeah yeah.

then i noticed that some of our actions are now taking forever. This is weird. I haven't changed any of our actions.

just now i look at our org settings > actions > runners: 

![image](https://user-images.githubusercontent.com/61225/198508194-09f65b53-1348-43c3-ba60-2c04cd04cf63.png)

then i look at org settings > actions > runners > ubuntu-latest-4-cores

and it looks like this: 
![image](https://user-images.githubusercontent.com/61225/198508925-5bf08386-b16f-46be-bb04-859a42d5097f.png)

that's really odd, because our actions minutes aren't anywhere near their limits: 
![image](https://user-images.githubusercontent.com/61225/198508979-93865169-85f7-4280-ab86-a52e5cd9f94e.png)

**Reply by petewalker** [reply]
@airtonix - The larger runners don't get paid for out of your entitlement minutes - they're always an added cost. See https://docs.github.com/en/actions/using-github-hosted-runners/using-larger-runners#understanding-billing

>**Note:** The larger runners do not use included entitlement minutes, and are not free for public repositories.
>
>Compared to standard GitHub-hosted runners, larger runners are billed differently. For more information, see "[Per-minute rates](https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions#per-minute-rates)".

**Comment by alexleonov-tactiq** [comment]
Same issue here - time spent on waiting for the runners completely negates the speed increase of the actual activities. Total times are actually longer than before.

**Comment by boardgamesaz** [comment]
This is definitely a great article. Thank you so much for taking the time to <a href="https://cookie-clicker.io/">cookie clicker</a> detail all of this for all of us. It was a great guide!

**Reply by selfagency** [reply]
![Spam-1](https://user-images.githubusercontent.com/2541728/199731667-031c5423-4f1d-40b5-a54f-9f9d71bc9103.jpeg)

**Comment by netdjw** [comment]
My experience is if I use this configuration
```yml
    strategy:
      matrix:
        node-version: [14.x]
```
Or this:
```yml
    strategy:
      fail-fast: false
      matrix:
        ci_node_total: [ 8 ]
        ci_node_index: [ 0, 1, 2, 3, 4, 5, 6, 7 ]
```
I see the `Waiting for a runner to pick up this job...` message, and nothing happens after 10 minutes wating.

But if the job don't use matrix strategy, it runs really fast.

Maybe this can help debug the issue.
